dbcon.js

```diff
@@ -7,11 +7,11 @@ var pool = mysql.createPool({
     dateStrings: 'true',
     typeCast: function castField(field, useDefaultTypeCasting) {
         if (field.type === "BIT" && field.length === 1) {
-            var bytes = field.buffer();
-            return (bytes[0] === 1);
+            var bytes = field.buffer()
+            return (bytes === null) ? null : (bytes[0] === 1)
```

Comment:
i noticed an error related to this come up sometimes. I wasn't sure how to deal with it. When you have time please point me in the direction to how you figured this out. I think we glossed over how this worked back in CS290 .